### PR TITLE
Fix misc TS types issues

### PR DESCRIPTION
**Pull request by kevinbarabash**
## Summary:
- make sure that the generated flow type use 'extends' instead of 'mixins'
- fix export types of i18n functions

Issue: None

## Test plan:
- yarn typecheck
- let CI run

### CODE REVIEW

File: .changeset/flat-clouds-sniff.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@khanacademy/wonder-blocks-i18n": patch
```
Comment (jeresig) [inline .changeset/flat-clouds-sniff.md:2]: I notice that only i18n is flagged as having changes, even though the build-script will almost certainly change all the packages (and require republishing them all).

File: build-scripts/gen-flow-types.ts [hunk 2]
```diff
@@ -14,6 +15,18 @@ for (const inFile of files) {
...
+        // so we do some post-processing on the files to clean that up.
+        const contents = fs.readFileSync(path.join(rootDir, outFile), "utf-8");
+        if (contents.includes(" mixins ")) {
+            console.log("replacing 'mixins' with 'extends'");
+            fs.writeFileSync(
+                path.join(rootDir, outFile),
+                contents.replace(/ mixins /g, " extends "),
+                "utf-8",
+            );
+        }
```
Comment (kevinbarabash) [inline build-scripts/gen-flow-types.ts:29]: I must've been looking at one of the earlier PRs instead what's on `main` in the wonder-stuff repo.

**Review by jeresig — APPROVED** [review]

File: packages/wonder-blocks-i18n/src/functions/i18n.ts [hunk 3]
```diff
@@ -233,8 +232,12 @@ export const $_: (
  * In Russian, any number ending in 1 (e.g. 21, 61) uses singular form.
  * You should instead still parameterize the number.
  */
-// @ts-expect-error [FEI-5019] - TS2322 - Type '(singular: string, plural: string, num: number | null | undefined, options: NGetOptions | undefined) => string' is not assignable to type 'ngettextOverloads'.
-export const ngettext: ngettextOverloads = (singular, plural, num, options) => {
```
Comment (kevinbarabash) [inline packages/wonder-blocks-i18n/src/functions/i18n.ts:237]: This was causing issues in the perseus repo when I tried to update it.

File: .changeset/flat-clouds-sniff.md (hunk 1 above)
Comment (kevinbarabash) [reply .changeset/flat-clouds-sniff.md:2]: Good catch.  I'll move the build-scripts into build-settings.

**Pull request merged**